scripts/dev_setup.sh

```diff
@@ -428,6 +436,28 @@ function install_cvc4 {
   rm -rf "$TMPFILE"
 }
 
+function install_codegen_deps {
+    if [[ "$PACKAGE_MANAGER" != "apt-get" ]]; then
+      # Not supported.
+      return
+    fi
+    echo "deb http://http.us.debian.org/debian/ buster-backports main" | "${PRE_COMMAND[@]}" tee -a /etc/apt/sources.list
+    echo "deb-src http://http.us.debian.org/debian/ buster-backports main" | "${PRE_COMMAND[@]}" tee -a /etc/apt/sources.list
+    "${PRE_COMMAND[@]}" apt-get update
+    "${PRE_COMMAND[@]}" apt-get upgrade -y
+    "${PRE_COMMAND[@]}" apt-get install -y apt-transport-https python3-all-dev python3-pip clang llvm default-jdk golang-1.14-go/buster-backports
+    "${PRE_COMMAND[@]}" ln -sf /usr/lib/go-1.14 /usr/lib/golang
+    "${PRE_COMMAND[@]}" python3 -m pip install pyre-check==0.0.59
```

Comment:
shouldn't the pips be handled as if they were proper versioned dependencies of the python project?   in a pip config file?    The pip content could then be added to the paths looked at by the caching in CI.    locally pip would just do it's thing?

Reply:
Pyre-check is a bit of exception as the tool that changes often (and becomes stricter and stricter?)
We don't need to pin numpy as much (>= 1.20 would be enough).

Reply:
I removed the constraint on numpy. Perhaps I should explain that the generator python3 code needs to be able to do `import numpy` and it "just works".